### PR TITLE
Loki: lexer was incorrectly handling negative numbers in a filter experssion

### PR DESCRIPTION
**Pull request by slim-bean**
Not sure if this is the correct way to handle this but the lexer saw the minus sign as a subtraction operation which was not expected in a filter expression.

### CODE REVIEW

File: pkg/logql/expr.y [hunk 1]
```diff
@@ -340,6 +340,13 @@ numberFilter:
     | IDENTIFIER NEQ NUMBER     { $$ = log.NewNumericLabelFilter(log.LabelFilterNotEqual, $1, mustNewFloat($3))}
     | IDENTIFIER EQ NUMBER      { $$ = log.NewNumericLabelFilter(log.LabelFilterEqual, $1, mustNewFloat($3))}
     | IDENTIFIER CMP_EQ NUMBER  { $$ = log.NewNumericLabelFilter(log.LabelFilterEqual, $1, mustNewFloat($3))}
+    | IDENTIFIER GT SUB NUMBER      { $$ = log.NewNumericLabelFilter(log.LabelFilterGreaterThan, $1, mustNewNegFloat($4))}
```
Comment (cyriltovena) [inline pkg/logql/expr.y:343]: I think it's cleaner if we make the lexer instead understand `NUMBER`.

Comment (cyriltovena) [reply pkg/logql/expr.y:343]: Which if I remember is what tried @dannykopping this way it will work everywere `NUMBER` is used.

Comment (dannykopping) [reply pkg/logql/expr.y:343]: Yeah I took the approach of replacing `NUMBER` with a token that could handle signed numbers:
https://github.com/grafana/loki/pull/3288/files

It was all working but I gave up trying to figure out how to parse `1+------1`, which is somewhat esoteric but I was concerned that it was representative of an underlying bug.

Comment (slim-bean) [reply pkg/logql/expr.y:343]: Happy to close this in favor of your PR danny, looks like you did more work as well.

`1+------1` isn't a case I think we need to handle? that's not a number it's a binary operation, we just want to handle numbers (including signed numbers)

**Comment by slim-bean** [comment]
Closing this in favor of #3288

**Pull request closed without merging**